File: 프로그래머스/2/42885. 구명보트/구명보트.py
# 80 70 50 50
from collections import deque

def solution(people, limit):
    answer = 0
    boat = deque(sorted(people, reverse = True))
    
    while boat:
        if len(boat) >= 2:
            if boat[0] + boat[-1] <= limit:
                boat.pop()
                boat.popleft()
            else:
                boat.popleft()
        else:
            boat.popleft()
        answer += 1
        
    return answer

File: 프로그래머스/2/42885. 구명보트/구명보트.js
function solution(people, limit) {
    let answer = 0;
    let left = 0;
    let right = people.length - 1;
    people = people.sort((a, b) => a - b);
    
    while (left <= right) {
        const curr = people[left] + people[right];
        if (curr > limit) right--;
        else if (curr <= limit) {
            right--;
            left++;  
        }
        
        answer++;
    }
    
    return answer;
}